src/common/utils/logger_helper.h

```diff
@@ -28,12 +28,41 @@ namespace LoggerHelpers
         return false;
     }
 
+    inline bool create_dir_if_does_not_exist(std::filesystem::path dir)
+    {
+        std::error_code err;
+        auto entry = std::filesystem::directory_entry(dir, err);
+        if (err.value())
+        {
+            Logger::error("Failed to create directory entry. {}", err.message());
+            return false;
+        }
+
+        if (!entry.exists())
+        {
+            Logger::warn("Directory {} does not exist", dir.string());
+            std::error_code err;
+            if (!std::filesystem::create_directory(dir, err))
+            {
+                Logger::error("Failed to create directory {}. {}", dir.string(), err.message());
+                return false;
+            }
+        }
+
+        return true;
+    }
+
     inline bool delete_other_versions_log_folders(std::wstring_view appPath, const std::filesystem::path& currentVersionLogFolder)
     {
         bool result = true;
         std::filesystem::path logFolderPath(appPath);
         logFolderPath.append(LogSettings::logPath);
 
+        if (!create_dir_if_does_not_exist(logFolderPath))
+        {
+            return false;
```

Comment:
I would expect that `false` is returned if the function fails. In this case there is nothing to delete, so it should return `true`

Reply:
Use dir_exists instead of create_dir_if_does_not_exist and return true if it does not exist.